### ISSUE TITLE
Support for non-english keyboards

### ISSUE THREAD

**Issue by sebastinas** [pwmt/girara#18]
*On GitLab by @Teu5us on Nov 16, 2019, 10:02*

---

I've been using Zathura for quite a long time now. Lack of support of Russian, in my case, is a little bit disappointing. It is said in zathura issue #693 that girara "fails to handle non-ascii keys". I tried using Emacs for viewing pdfs but zathura is what I'm now used to.

So I added Russian symbols to callbacks.c where numpad keys are mapped to numbers row keys (line 42 and further):

```
  switch (*keyval) {
    ...
    case GDK_KEY_KP_9:
      *keyval = GDK_KEY_9;
      break;
    case GDK_KEY_Cyrillic_shorti:
      *keyval = GDK_KEY_q;
      break;
    case GDK_KEY_Cyrillic_SHORTI:
      *keyval = GDK_KEY_Q;
      break;
    ...
  }
```

It actually seems to work just like in Vim, where the letters work fine, but some punctuation marks have to be used based on active keyboard layout. I think this can also work with umlauts and other symbols not present on english keyboards. Would love to see something like this implemented.

**Comment by sebastinas** [comment]
*On GitLab by @PRESFIL on Nov 1, 2020, 08:49*

---

You can made some layout sensitive modes like in Vim. For example, in `Command mode` keyboard layout a valuable, but in `Normal`|`Fullscreen`|`Presentation`|`Index` modes (all of them are variations of `Normal mode`) keyboard layout are not valuable (because only in `Command mode` you can input everything, like file path).

In Vim user [can invoke][2] mappings by `<C-^>`, specified in [`'keymap'`][1]. But there `'keymap'` specified manually. May be it must be gathered automatically from Xorg|Wayland compositor.

[1]: http://vimdoc.sourceforge.net/htmldoc/options.html#'keymap'
[2]: http://vimdoc.sourceforge.net/htmldoc/options.html#'lmap'

**Comment by sebastinas** [comment]
mentioned in issue zathura#315

**Comment by sebastinas** [comment]
*On GitLab by @roman on Jan 6, 2023, 18:49*

---

I confirm that non-latin characters can not be mapped:
```
map ← scroll left
map ↓ scroll down
map ↑ scroll up
map → scroll right
```
gives no effect when pressing the keys that produce `←↓↑→` symbols on my keyboard.

Come on, no progress for 3 years?

**Comment by sebastinas** [comment]
Fixed in 7eebe2b91b9fdd5b39dc3d0aefb1e4b08218788b
